raster/r.in.pdal/info.cpp

```diff
@@ -52,43 +61,153 @@ void get_extent(struct StringList *infiles, double *min_x, double *max_x,
         catch (const std::exception &err) {
             G_fatal_error(_("PDAL error: %s"), err.what());
         }
+        spatial_reference = las_reader.getSpatialReference();
+        std::string dataset_wkt = spatial_reference.getWKT();
+        bool proj_match = is_wkt_projection_same_as_loc(dataset_wkt.c_str());
+
         const pdal::LasHeader &las_header = las_reader.header();
+
+        min_x_ = las_header.minX();
+        min_y_ = las_header.minY();
+        min_z_ = las_header.minZ();
+        max_x_ = las_header.maxX();
+        max_y_ = las_header.maxY();
+        max_z_ = las_header.maxZ();
+#ifdef R_IN_PDAL_USE_NOSRS
+        bool need_to_reproject = !nosrs && !proj_match &&
+                                 spatial_reference.valid() &&
+                                 !spatial_reference.empty();
+#else
+        bool need_to_reproject = !proj_match && spatial_reference.valid() &&
+                                 !spatial_reference.empty();
+#endif
+
+        if (need_to_reproject) {
+            get_reprojected_extent(spatial_reference, &min_x_, &max_x_, &min_y_,
+                                   &max_y_, &min_z_, &max_z_);
+        }
         if (first) {
-            *min_x = las_header.minX();
-            *min_y = las_header.minY();
-            *min_z = las_header.minZ();
-            *max_x = las_header.maxX();
-            *max_y = las_header.maxY();
-            *max_z = las_header.maxZ();
-
-            first = 0;
+            *min_x = min_x_;
+            *min_y = min_y_;
+            *min_z = min_z_;
+            *max_x = max_x_;
+            *max_y = max_y_;
+            *max_z = max_z_;
+
+            first = false;
+        }
+        else {
+            if (*min_x > min_x_)
+                *min_x = min_x_;
+            if (*min_y > min_y_)
+                *min_y = min_y_;
+            if (*min_z > min_z_)
+                *min_z = min_z_;
+            if (*max_x < max_x_)
+                *max_x = max_x_;
+            if (*max_y < max_y_)
+                *max_y = max_y_;
+            if (*max_z < max_z_)
+                *max_z = max_z_;
+        }
+    }
+}
+/* Reproject 4 points representing bounding box using PDAL pipeline */
+void get_reprojected_extent(pdal::SpatialReference &spatial_reference,
+                            double *min_x, double *max_x, double *min_y,
+                            double *max_y, double *min_z, double *max_z)
+{
+    pdal::PointTable table;
+    table.layout()->registerDim(pdal::Dimension::Id::X);
+    table.layout()->registerDim(pdal::Dimension::Id::Y);
+    table.layout()->registerDim(pdal::Dimension::Id::Z);
+
+    pdal::PointViewPtr view(new pdal::PointView(table));
+    pdal::PointId idx = 0;
+    int points_per_edge = 10;
+    // Generate points along the 4 edges of the bounding box
+    // Add single corner point and interior points for each edge
```

Comment:
Do you want to add a comment for the lazy ones how the single corner point is ensured? (like "first point of each edge starts at the corner and the last point is right before the corner").